### PR TITLE
custom json methods for virtual machine

### PR DESCRIPTION
**Pull request by baude**
Add custom JSON methods for Marshal and Unmarshal so other projects can write a configuration file based on virtual machine.  the key issue is the use of interfaces, for example devices, in virtual machine.  When using JSON to marshal and unmarshal, we need to tag what kind of device it is so the process can tag and ultimately re-identify the proper device.

### CODE REVIEW

File: pkg/config/json.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+		Kind vmComponentKind `json:"kind"`
+		LinuxBootloader
+	}
+	return json.Marshal(blWithKind{
+		Kind:            linuxBootloader,
+		LinuxBootloader: *bootloader,
+	})
+}
+
+func (dev *VirtioNet) MarshalJSON() ([]byte, error) {
```
Comment (jakecorrenti) [inline pkg/config/json.go:173]: Will need to add functions for `VirtioInput` and `VirtioGPU`

**Comment by jakecorrenti** [comment]
@cfergeau friendly ping

**Comment by baude** [comment]
rebased and updated for virtioinput and virtiogpu

@jakecorrenti ptal

**Comment by jakecorrenti** [comment]
One small change, but lgtm otherwise

**Comment by cfergeau** [comment]
Closing this in favour of https://github.com/crc-org/vfkit/pull/46 where I did some follow-up work.

**Pull request closed without merging**